src/HttpHeader.h

```diff
@@ -191,14 +194,6 @@ int httpHeaderParseQuotedString(const char *start, const int len, String *val);
 SBuf httpHeaderQuoteString(const char *raw);
 
 void httpHeaderCalcMask(HttpHeaderMask * mask, Http::HdrType http_hdr_type_enums[], size_t count);
-
-inline bool
-HttpHeader::chunked() const
-{
-    return has(Http::HdrType::TRANSFER_ENCODING) &&
```

Comment:
I am very worried that we are replacing this always-fresh method with a cached-by-parse() value while manually adding Transfer-Encoding headers to HttpHeader, completely bypassing parse(). I suggest keeping this method and just changing its implementation to a single has() call because the parser will remove non-chunked Transfer-Encoding values and we never add non-chunked Transfer-Encoding values.

This change request supersedes teChunked polishing change requests.

Reply:
I do not believe there is ever a valid case for adding/changing these framing headers to values different from those seen by the parser. Doing so would mean the payload is different to the one the header represents.

Reply:
I will try to rephrase: The existence of an API for adding a Transfer-Encoding:chunked header is incompatible with the chunked() implementation in the PR. The official implementation does not have that problem. We should not create such a problem without a compelling reason, especially when Squid already uses that API. I do not see such a reason.

To see cases where these framing headers are added without going through the parser use `git grep putStr.*TRANSFER_ENCODING`.

Reply:
So what exactly did you mean then when you argued for implementing this function as a flag test?

Reply:
I was thinking about a (correct) implementation that would not raise these red flags. Such an implementation may require changing some putStr() callers as well. However, that idea was meant for the rather different Transfer-Encoding parsing code that no longer exists.

Given the current PR code, returning has() is probably the best approach for chunked(): It is simple, reliable, fast enough, and does not conflict with Squid code that adds chunked encoding (without any caller modifications). It would require deleting unsupported Transfer-Encoding headers, but deleting them is probably a good idea anyway (just like we already delete bad Content-Length headers and Transfer-Encoding:identity headers).

Reply:
Okay. If we are going to assume putStr() API usage, that caller may set a non-chunked coding so I am leaving the check for "chunked" it used to do (in aff8e63).

That leaves some potentially nasty issues with custom unsupported/invalid T-E not being detected. But since it is coming from an "internal" source I think we can (should?) leave that as a problem for the admin or ICAP/eCAP service dev to ensure never happens - for now at least.